### PR TITLE
Add to google cal

### LINKED ISSUE #517: add_to_google_cal field needs to be added to medication_model
Currently, add_to_google_cal is simply an attr_accessor on the medication model. Because of this whether a calendar reminder was added is not saved to the model, and it also makes doing comparisons on this field difficult.

### PR DESCRIPTION
**Pull request by abrophy**
resolves #517

### CODE REVIEW

File: db/migrate/20170830075513_add_add_to_google_cal_to_medication.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddAddToGoogleCalToMedication < ActiveRecord::Migration[5.0]
+  def change
+    add_column :medications, :add_to_google_cal, :boolean
```
Comment (alexiamcdonald) [inline db/migrate/20170830075513_add_add_to_google_cal_to_medication.rb]: Perhaps add default to boolean?

Comment (abrophy) [reply db/migrate/20170830075513_add_add_to_google_cal_to_medication.rb]: sure, right on, better to be explicit

**Comment by alexiamcdonald** [comment]
This PR is having some changes done. Working with @abrophy to make it better. Thought we should let everyone else know what's happening. 😄

File: app/models/moment.rb [hunk 2]
```diff
@@ -17,6 +17,7 @@
 #  comment    :boolean
 #  strategy   :text
 #  slug       :string
+#  sentiment  :float
```
Comment (julianguyen) [inline app/models/moment.rb]: I think this was accidentally committed

File: spec/models/moment_spec.rb [hunk 3]
```diff
@@ -15,6 +15,7 @@
 #  comment    :boolean
 #  strategy   :text
 #  slug       :string
+#  sentiment  :float
```
Comment (julianguyen) [inline spec/models/moment_spec.rb]: I think this was accidentally committed too.

File: app/helpers/calendar_helper.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+       medication.refill &&
+       (medication.add_to_google_cal_changed? || medication.refill_changed?)
+      true
+    else
+      false
+    end
+  end
+
+  def new_cal_refill_reminder_unneeded?(medication)
+    if medication.add_to_google_cal &&
```
Comment (julianguyen) [inline app/helpers/calendar_helper.rb]: Could replace the contents of this method with just `!new_cal_refill_reminder_needed?(medication)`

File: app/helpers/medication_refill_helper.rb [hunk 5]
```diff
@@ -1,10 +1,11 @@
 # frozen_string_literal: true
 
 module MedicationRefillHelper
+  include CalendarHelper
   # Save refill date to Google calendar
   def save_refill_to_google_calendar(medication)
     return true unless current_user.google_oauth2_enabled? &&
-                       medication.add_to_google_cal == '1'
+                       new_cal_refill_reminder_needed?(medication)
```
Comment (julianguyen) [inline app/helpers/medication_refill_helper.rb]: 🎉  yassss

**Pull request merged**